src/publisher/flow-control.ts

```diff
@@ -0,0 +1,197 @@
+/*!
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as defer from 'p-defer';
+
+/**
+ * @typedef PublisherFlowControlAction Options for flow control actions
+ *
+ * @property {number} Ignore Ignore all flow control; don't take any action
+ *     based on outstanding requests.
+ * @property {number} Pause When flow control limits are exceeded, clients
```

Comment:
Yeah, I was a little bit weird with "block" simply because you really don't block in Node (it has a different meaning). I can make it block if we want to be consistent, though.